DEVELOPMENTGUIDE.md

```diff
@@ -0,0 +1,78 @@
+# Development Guide
+The Development Guide establishes conventions that regular contributors are expected to follow on ScubaGoggles. If you are a new or first-time contributor, you are not required to follow all these conventions, though it is certainly appreciated. See instead [CONTRIBUTING.md](CONTRIBUTING.md) for licencing details and guidance for new contributors.
+
+## Branching Strategy
+We adhere to [GitHub flow](https://docs.github.com/en/get-started/using-github/github-flow) as closely as our specific contstraints allow. In general, this means:
+- Development is performed on short-lived features branches (see [Contibuting Code](#contributing-code) for more details)
+- There is only one primary branch, main. After review, changes are committed directly to main, rather than to long-lived develop or release branches.
+
+There are exceptions to this, necessitated by the extensive review process the baseline documents themselves undergo, during which no baseline changes can be made.
+To accommodate this, the development lifecycle on ScubaGoggles is as follows:
+![image](/docs/images/scubagoggles_lifecycle.svg)
+
+
+Baseline submission triggers a baseline freeze, at which point the following actions need to be performed:
+1. Bump the code in main to the upcoming version (to reduce merge conflicts downstream)
+2. Make a note of the last commit hash, to help in case of complications rebasing down the line.
+3. Create a temporary protected branch dedicated to baseline changes.
+
+A new release triggers the transision back to normal operations. At this time, the following actions need to be performed:
+1. Pause development on main and perform functional testing
+2. Assuming the testing results are satisfactory, create the release off of main
+3. Rebase the temporary baseline branch with main
+4. Implement the baseline code issues created during the baseline freeze to get the release branch into a fully functional state
+5. Merge the baseline branch into main
+6. Delete the baseline branch
+
+## Branching Structure
+See [Branching Structure Diagram](BRANCHINGSTRUCTURE.md) for a visual representation of the branch structure, both during a code freeze and otherwise.
+
+## Creating Issues
+All needed changes (e.g., new features, bug fixes) need to be tracked in an issue. When creating an issue:
+- Be sure to follow the template
+- Add the appropriate labels (e.g., "bug," "enhancement")
+- Add the issue to the correct project
+
+## Contributing Code
+### Step 1: Create a branch and implement changes
+1. Open the corresponding issue. If no issue has been created for the intended change, first create one (see [Creating Issues](#creating-issues)).
+2. Add yourself as the assignee on the issue. NOTE: Having just one assignee per issue is preferred.
+3. Add the issue to the current milestone if it hasn't already been added.
```

Comment:
I agree about that. But this is in the "Contributing Code" section, not the "Creating Issues" section, i.e., this is the point when you're about to sit down and implement the issue. If it hasn't been added to the milestone yet, it should be, because you're now actively working it. Though one could argue that not all active work will be targeted to the current milestone, but I think that will almost always be the case here.

Reply:
So include this under the "Creating Issues" section? Something like:
```
3. Add the issue to the correct project
4. Leave the milestone option blank, new issues will be triaged during sprint planning and assigned to an appropriate milestone.
```